pkg/client/mapper.go

```diff
@@ -103,3 +103,57 @@ func NewBidirectionalNewOneToManyGroupMapper(source, dest, groupMappingFile stri
 	}
 	return nil, nil, fmt.Errorf("unsupported source to dest mapper type: source %s, dest %s", source, dest)
 }
+
+// UserMapperImpl implements groupsync.UserMapper.
+type UserMapperImpl map[string]string
+
+func (u UserMapperImpl) MappedUserID(ctx context.Context, userID string) (string, error) {
+	v, ok := u[userID]
+	if !ok {
+		return "", fmt.Errorf("no mapped user for %s", userID)
```

Comment:
Can you return this error?

https://github.com/abcxyz/team-link/blob/main/pkg/groupsync/errors.go#L24

There is special logic to handle this error here: https://github.com/abcxyz/team-link/blob/main/pkg/groupsync/manytomanysyncer.go#L212

Reply:
Changed to return `groupsync.ErrTargetUserIDNotFound`